### PR TITLE
refactor(telegrafreceiver): refactor

### PR DESCRIPTION
**Pull request by pmalek-sumo**
~~This will allow something like this:~~

```yaml
processors:
  metricstransform:
    transforms:
    - include: Net_.*
      match_type: regexp
      action: update
      operations:
      - action: update_label
        label: interface
        new_label: Interface
```

~~to work.~~

**EDIT**: Let's leave the functionality as is (i.e. retain copying the telegraf tags to resource attributes) but let's stick to the refactor that has been done in this PR to allow easier changes in code in the future and to decrease code duplication related to metric translation.

### CODE REVIEW

File: pkg/receiver/telegrafreceiver/converter.go [hunk 1]
```diff
@@ -181,15 +133,78 @@ func (mc metricConverter) Convert(m telegraf.Metric) (pdata.Metrics, error) {
 	return ms, nil
 }
 
-func setName(pm *pdata.Metric, name string, key string, separateField bool) {
-	if separateField {
-		pm.SetName(name)
+func (mc metricConverter) convertToGauge(name string, f *telegraf.Field, opts ...MetricOpt) (pdata.Metric, error) {
```
Comment (pmatyjasek-sumo) [inline pkg/receiver/telegrafreceiver/converter.go:136]: please add a comment

File: pkg/receiver/telegrafreceiver/converter.go [hunk 2]
```diff
@@ -181,15 +133,78 @@ func (mc metricConverter) Convert(m telegraf.Metric) (pdata.Metrics, error) {
...
+		pm = newIntGauge(vv, opts...)
+
+	default:
+		return pm, fmt.Errorf("unsupported underlying type: %T", v)
+	}
+
+	return pm, nil
+}
+
+func (mc metricConverter) convertToCounter(name string, f *telegraf.Field, opts ...MetricOpt) (pdata.Metric, error) {
```
Comment (pmatyjasek-sumo) [inline pkg/receiver/telegrafreceiver/converter.go:166]: same as above

File: pkg/receiver/telegrafreceiver/converter.go [hunk 3]
```diff
@@ -181,15 +133,78 @@ func (mc metricConverter) Convert(m telegraf.Metric) (pdata.Metrics, error) {
...
+		pm = newIntCounter(vv, opts...)
+
+	default:
+		return pm, fmt.Errorf("unsupported underlying type: %T", v)
+	}
+
+	return pm, nil
+}
+
+func (mc metricConverter) createMetricName(name string, key string) string {
```
Comment (pmatyjasek-sumo) [inline pkg/receiver/telegrafreceiver/converter.go:196]: comment

File: pkg/receiver/telegrafreceiver/converter.go (hunk 1 above)
Comment (pmalek-sumo) [reply pkg/receiver/telegrafreceiver/converter.go:136]: done

File: pkg/receiver/telegrafreceiver/converter.go (hunk 3 above)
Comment (pmalek-sumo) [reply pkg/receiver/telegrafreceiver/converter.go]: done

File: pkg/receiver/telegrafreceiver/converter.go (hunk 2 above)
Comment (pmalek-sumo) [reply pkg/receiver/telegrafreceiver/converter.go]: done

**Review by sumo-drosiek — APPROVED** [review]
LGTM

**Pull request merged**